metric/metric_recorder.go

```diff
@@ -0,0 +1,44 @@
+package metric
+
+import (
+	"context"
+	"net/http"
+	"strings"
+
+	"github.com/fabric8-services/fabric8-wit/log"
+	"github.com/goadesign/goa"
+	"github.com/prometheus/client_golang/prometheus"
+)
+
+var reqLabels = []string{"entity", "action"}
+
+// Recorder record metrics
+func Recorder() goa.Middleware {
+	reqCnt := prometheus.NewCounterVec(prometheus.CounterOpts{
+		Name: "wit_requests_total",
+		Help: "Total number of WIT requests.",
+	}, reqLabels)
+	// TODO VN: Check err, need prometheus version change.
+	prometheus.Register(reqCnt)
```

Comment:
Current version of client_golang/prometheus is old so few func/type not available for error handling.
**Que:** Shall I change to use latest stable version?

Reply:
Yes, upgrade to latest stable